RDBM/XML.h

```diff
@@ -0,0 +1,285 @@
+		{
+			TiXmlElement *value = new TiXmlElement("Value");
+			record = new TiXmlElement("Record");
+			for (unsigned int i = 0; i < source.GetCName().size(); i++)
+			{
+				if (ctype[i] == 'i')
+				{
+					temp = source.GetRecord(j)->record[i]->Getv();
+					int val = *(int*)temp;
+					value->LinkEndChild(new TiXmlText(std::to_string(val).c_str()));
+					record->LinkEndChild(value);
+					if (i != source.GetCName().size() - 1)
+					{
+						value = new TiXmlElement("Value");
+					}
+				}
+				else if (ctype[i] == 'd')
+				{
+					temp = source.GetRecord(j)->record[i]->Getv();
+					double val = *(double*)temp;
+					value->LinkEndChild(new TiXmlText(std::to_string(val).c_str()));
+					record->LinkEndChild(value);
+					if (i != source.GetCName().size() - 1)
+					{
+						value = new TiXmlElement("Value");
+					}
+				}
+				else if (ctype[i] == 's')
+				{
+					temp = source.GetRecord(j)->record[i]->Getv();
+					std::string val = *(std::string*)temp;
+					value->LinkEndChild(new TiXmlText(val.c_str()));
+					record->LinkEndChild(value);
+					if (i != source.GetCName().size() - 1)
+					{
+						value = new TiXmlElement("Value");
+					}
+				}
+			}
+			table->LinkEndChild(record);
+		}
+	}
+	doc.SaveFile("dbdata.xml");
+}
+
+Table* BuildTable(Table& tb)
+{
+	TiXmlDocument doc;
+	doc.LoadFile("dbstructure.xml");
+	std::vector<std::string> names;
+	std::vector<char> types;
+	
+	TiXmlElement *root = doc.RootElement();
+	if (NULL != root)
+	{
+		TiXmlElement *table = root->FirstChildElement("Table");
+		if (NULL != table)
+		{
+			while (table)
+			{
+				TiXmlElement *record = table->FirstChildElement("Record");
+				while (record)
+				{
+					TiXmlElement *type = record->FirstChildElement("Type");
+					std::cout << type->GetText() << " ";
+					TiXmlElement *name = record->FirstChildElement("Name");
+					std::cout << name->GetText() << " ";
+					names.push_back(name->GetText());
+					std::string stype = type->GetText();
+					if (stype == "Integer")
+						types.push_back('i');
+					else if (stype == "Double")
+						types.push_back('d');
+					else if (stype == "String")
+						types.push_back('s');
+					record = record->NextSiblingElement();
+				}
+				std::cout << std::endl;
+				table = table->NextSiblingElement();
+			}
+		}
+	}
+	tb.Create(names, types);
+	doc.LoadFile("dbdata.xml");
+	root = doc.RootElement();
+	if (NULL != root)
+	{
+		TiXmlElement *table = root->FirstChildElement("Table");
+		if (NULL != table)
+		{
+			while (table)
+			{
+				TiXmlElement *record = table->FirstChildElement("Record");
+				while (record)
+				{
+					TiXmlElement *value = record->FirstChildElement("Value");
+					while (value)
+					{
+						std::cout << value->GetText() << " ";
+						value = value->NextSiblingElement();
+					}
+					record = record->NextSiblingElement();
+					std::cout << std::endl;
+				}
+				std::cout << std::endl;
+				table = table->NextSiblingElement();
+			}
+		}
+	}
+	return nullptr;
```

Comment:
This is the only return statement I've found. Do we return nullptr always?

Reply:
no. nullptr was put there when function was just created and was empty.when it would be complete there will be correct return